### PR TITLE
CFY-5413 Improved dsl exception message in deployment update

### PR DESCRIPTION
**Pull request by AviaE**

### CODE REVIEW

File: rest-service/manager_rest/resources_v2_1.py [hunk 1]
```diff
@@ -146,10 +145,9 @@ def _prepare_and_submit_blueprint(cls,
                     shutil.copy(source_file, dest_file)
 
             return update
-        except DslParseException, ex:
+        except BaseException:
```
Comment (AviaE) [inline rest-service/manager_rest/resources_v2_1.py]: BaseException --> Exception

File: rest-service/manager_rest/deployment_update/manager.py [hunk 2]
```diff
@@ -105,18 +106,22 @@ def stage_deployment_update(self,
...
+                resources_base_url=file_server_base_url,
+                additional_resources=[blueprint_resource_dir],
+                **app_context.get_parser_context())
+            # Updating the new inputs with the deployment inputs
+            # (overriding old values and adding new ones)
+            inputs = copy.deepcopy(deployment.inputs)
+            inputs.update(additional_inputs)
+
+            # applying intrinsic functions
+            prepared_plan = tasks.prepare_deployment_plan(plan, inputs=inputs)
```
Comment (AviaE) [inline rest-service/manager_rest/deployment_update/manager.py]: prepare_deployment_plan may raise additional error types

File: rest-service/manager_rest/deployment_update/manager.py [hunk 3]
```diff
@@ -105,17 +106,29 @@ def stage_deployment_update(self,
...
-        # values and adding new ones)
-        inputs = copy.deepcopy(deployment.inputs)
-        inputs.update(additional_inputs)
+        try:
+            plan = tasks.parse_dsl(
+                app_path,
+                resources_base_url=file_server_base_url,
+                additional_resources=[blueprint_resource_dir],
+                **app_context.get_parser_context())
+            # Updating the new inputs with the deployment inputs
```
Comment (AviaE) [inline rest-service/manager_rest/deployment_update/manager.py]: Not in the `try`s

File: rest-service/manager_rest/deployment_update/manager.py [hunk 4]
```diff
@@ -105,17 +106,29 @@ def stage_deployment_update(self,
...
+            # (overriding old values and adding new ones)
+            inputs = copy.deepcopy(deployment.inputs)
+            inputs.update(additional_inputs)
+
+        except parser_exceptions.DSLParsingException as ex:
+            raise manager_rest.manager_exceptions.InvalidBlueprintError(
```
Comment (AviaE) [inline rest-service/manager_rest/deployment_update/manager.py]: Inconsistent with the other two manager exceptions.
Order the imports

File: rest-service/manager_rest/test/test_deployment_updates.py [hunk 5]
```diff
@@ -33,6 +34,47 @@ def test_get_empty(self):
...
+        deployment_id = 'dep'
+        self._deploy_base(deployment_id, 'no_output.yaml')
+
+        with patch('dsl_parser.tasks.parse_dsl') as parse_dsl_mock:
+            parse_dsl_mock.side_effect = \
+                manager_exceptions.InvalidBlueprintError()
+            # # It doesn't matter that we are updating the deployment with the
+            # same blueprint, since we mocked the blueprint parsing process.
+            response = self._update(deployment_id, 'no_output.yaml')
+            self.assertEquals(response.status_code, 400)
```
Comment (AviaE) [inline rest-service/manager_rest/test/test_deployment_updates.py]: expected on the left

File: rest-service/manager_rest/test/test_deployment_updates.py [hunk 6]
```diff
@@ -33,6 +34,47 @@ def test_get_empty(self):
         result = self.client.deployment_updates.list()
         self.assertEquals(0, len(result))
 
+    def test_invalid_blueprint_raises_invalid_blueprint_exception(self):
+        deployment_id = 'dep'
+        self._deploy_base(deployment_id, 'no_output.yaml')
+
+        with patch('dsl_parser.tasks.parse_dsl') as parse_dsl_mock:
+            parse_dsl_mock.side_effect = \
+                manager_exceptions.InvalidBlueprintError()
```
Comment (AviaE) [inline rest-service/manager_rest/test/test_deployment_updates.py]: Change to the correct dsl exception

File: rest-service/manager_rest/test/test_deployment_updates.py [hunk 7]
```diff
@@ -33,6 +34,47 @@ def test_get_empty(self):
...
+            # # It doesn't matter that we are updating the deployment with the
+            # same blueprint, since we mocked the blueprint parsing process.
+            response = self._update(deployment_id, 'no_output.yaml')
+            self.assertEquals(response.status_code, 400)
+            self.assertEquals(response.json['error_code'],
+                              'invalid_blueprint_error')
+
+    def test_raising_input_errors(self):
```
Comment (AviaE) [inline rest-service/manager_rest/test/test_deployment_updates.py]: Split into two tests

File: rest-service/manager_rest/deployment_update/manager.py [hunk 8]
```diff
@@ -13,25 +13,22 @@
...
 from datetime import datetime
-from os import path
 
 from flask import current_app
 
 from dsl_parser import constants, tasks
 from dsl_parser import exceptions as parser_exceptions
-from manager_rest.deployment_update import step_extractor
-import manager_rest.manager_exceptions
-import manager_rest.workflow_client as wf_client
```
Comment (ran-z) [inline rest-service/manager_rest/deployment_update/manager.py]: no need for line break here

**Comment by AviaE** [comment]
Reviewed by @ran-z

**Pull request merged**